micall/tests/test_consensus_aligner.py

```diff
@@ -784,20 +785,248 @@ def test_write_alignment(projects):
 # noinspection DuplicatedCode
 def test_write_multiple_alignments(projects):
     alignments_file = StringIO()
-    aligner = ConsensusAligner(projects, alignments_file=alignments_file)
+    aligner = ConsensusAligner(projects, alignments_file=alignments_file, contig_name='test_contig')
     aligner.coordinate_name = "test"
     amino_alignments = [AminoAlignment(0, 10, 20, 30, 0, 1, aligned_query="ATA", aligned_ref="ACA", ref_amino_start=10)]
     aligner.write_alignments_file(amino_alignments, aligner.alignments_writer)
-    aligner2 = ConsensusAligner(projects, alignments_file=alignments_file)
+    aligner2 = ConsensusAligner(projects, alignments_file=alignments_file, contig_name='test_contig_2')
     aligner2.coordinate_name = "2ndtest"
     amino_alignments2 = [AminoAlignment(0, 20, 40, 60, 0, 2, aligned_query="ABC", aligned_ref="DEF", ref_amino_start=4)]
     aligner2.write_alignments_file(amino_alignments2, aligner2.alignments_writer)
 
     expected_text = """\
-coordinate_name,action,query_start,query_end,ref_start,ref_end,reading_frame,\
+coordinate_name,contig,action,query_start,query_end,ref_start,ref_end,reading_frame,\
 ref_amino_start,aligned_query,aligned_ref
-test,MATCH,0,10,20,30,1,10,ATA,ACA
-2ndtest,MATCH,0,20,40,60,2,4,ABC,DEF
+test,test_contig,MATCH,0,10,20,30,1,10,ATA,ACA
+2ndtest,test_contig_2,MATCH,0,20,40,60,2,4,ABC,DEF
 """
 
     assert alignments_file.getvalue() == expected_text
+
+
+# noinspection DuplicatedCode
+def test_count_coord_concordance(projects):
+    aligner = ConsensusAligner(projects)
+    aligner.consensus = "AGATTTCGATGATTCAGAAGATTTGCA"
+    coord_ref = "AGATTTCGATGATTCAGAAGATAAGCA"
+    # changed nucs:                    ^^
+    aligner.alignments = [AlignmentWrapper(r_st=0, r_en=27, q_st=0, q_en=27, cigar=[[27, CigarActions.MATCH]])]
+
+    expected_concordance_list = [0.0]*10 + [1.0, 1.0, 1.0, 0.95, 0.9, 0.9, 0.9, 0.9] + [0.0]*9
+
+    with patch('micall.core.project_config.ProjectConfig.getGenotypeReference') as mock_get_reference:
+        mock_get_reference.return_value = coord_ref
+        concordance_list = aligner.coord_concordance()
+
+    assert concordance_list == expected_concordance_list
+
+
+# noinspection DuplicatedCode
+def test_count_coord_concordance_short_match(projects):
+    aligner = ConsensusAligner(projects)
+    aligner.consensus = "AGATTTCGATGATTCAGAAGATTTGCA"
+    coord_ref = "AGATTTCGATGATTCAGAAGATTTGCATTT"
+    aligner.alignments = [AlignmentWrapper(r_st=0, r_en=15, q_st=0, q_en=15, cigar=[[15, CigarActions.MATCH]])]
+
+    expected_concordance_list = [0.0]*10 + [0.75, 0.7, 0.65, 0.6, 0.55, 0.5, 0.45, 0.4, 0.35, 0.3, 0.25] + [0.0]*9
```

Comment:
Can you explain what's happening here?

Reply:
Added a comment explaining what's going on